types/array.prototype.flatmap/auto.d.ts

```diff
@@ -1,6 +1,13 @@
 interface Array<T> {
     flatMap<U, R extends object | undefined = undefined>(
-        fn: (this: R, x: T, index: number, array: this) => U[],
+        fn: (this: R, x: T, index: number, array: this) => ReadonlyArray<U>,
```

Comment:
Does `ReadonlyArray<U>` already include `U[]`?

Reply:
I tested to be sure that returning a read-write array also works. And there are a number of existing tests in `array.prototype.flatmap-tests.ts` that I think also verify that.

Reply:
Yes, I think `ReadonlyArray` is just a subset of `Array`, or something close enough to that. Anyway, a plain array is assignable to a `ReadonlyArray`.